solr/solrj/src/java/org/apache/solr/common/util/ResumableInputStream.java

```diff
@@ -84,6 +83,34 @@ public int read(boolean isRetry) throws IOException {
     return val;
   }
 
+  @Override
+  public int read(byte[] b, int off, int len) throws IOException {
+    return read(b, off, len, false);
+  }
+
+  public int read(byte[] b, int off, int len, boolean isRetry) throws IOException {
+    if (delegate == null) {
+      delegate = nextInputStreamSupplier.apply(bytesRead);
+    }
+    int readLen;
+    try {
+      readLen = delegate.read(b, off, len);
+      if (len >= 0) {
```

Comment:
should be readLen rather than len?

Reply:
Yep, dumb miss.

And unfortunately this is tested:
```java
// test both read() and read(buffer, off, len)
if (random().nextBoolean()) {
  done = input.read() == -1;
  if (!done) {
    byteCount++;
  }
} else {
  int readLen = input.read(buffer, 0, random().nextInt(maxBuffer) + 1);
  if (readLen > 0) {
    byteCount += readLen;
  } else {
    // We are done when readLen = -1
    done = true;
  }
}
```

Reply:
Ahh the reason it's not a big deal is that we never call read() after getting a -1, so the `byteCount` isn't really used. So it's wrong, but our test (and most usages of InputStream) didn't see any issue, because we follow the contract of InputStream.